javalang/parser.py

```diff
@@ -1769,7 +1794,13 @@ def parse_expression_2_rest(self):
     @parse_debug
     def parse_expression_3(self):
         prefix_operators = list()
-
+        if self.would_accept('('):
+            try:
+                lambda_exp = self.parse_lambda_expression()
```

Comment:
Could you move the `with self.tokens:` here with the try/except? This will be more consistent with the uses of this pattern in this file.